src/MyEventEmitter.js

```diff
@@ -1,12 +1,85 @@
 'use strict';
 
-export class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+class MyEventEmitter {
+  constructor() {
+    this.events = {};
+  }
+
+  on(eventName, listener) {
+    if (typeof listener !== 'function') {
+      throw new Error('Listener must be a function!');
+    }
+
+    this.events[eventName] = this.events[eventName] || [];
```

Comment:
You can use destructuring for events in all alike cases.
It can look like:
const { events } = this;
And further use:
events[eventName]

Reply:
agree